app/aws/PasskeyDB.scala

```diff
@@ -218,4 +218,54 @@ object PasskeyDB {
       )
     )
   )
+
+  def loadCredentialByName(
+      user: UserIdentity,
+      passkeyName: String
+  )(implicit dynamoDB: DynamoDbClient): Try[QueryResponse] =
+    Try {
+      val expressionValues = Map(
+        ":username" -> AttributeValue.fromS(user.username),
+        ":passkeyName" -> AttributeValue.fromS(passkeyName)
+      )
+      val request = QueryRequest
+        .builder()
+        .tableName(tableName)
+        .keyConditionExpression("username = :username")
+        .filterExpression("passkeyName = :passkeyName")
+        .expressionAttributeValues(expressionValues.asJava)
+        .build()
+      dynamoDB.query(request)
+    }.recoverWith(err =>
+      Failure(JanusException.failedToLoadDbItem(user, tableName, err))
+    )
+
+  def delete(
+      user: UserIdentity,
+      passkeyName: String
```

Comment:
The trouble with deleting by name is that it's potentially mutable, a user could have multiple passkeys with the same name, and the fact that they might consist of multiple words is potentially difficult.
We know that the user name and credential ID are unique so it would be better to include the passkey credential IDs in the HTML template when we show the user account page and then delete by user name and credential ID.

Reply:
Addressed in https://github.com/guardian/janus-app/pull/571/commits/1e7e159ecd4789e17c929599c2ea165043056436

Reply:
Addressed in https://github.com/guardian/janus-app/pull/571/commits/941fcbc2253f2eb1c5de5f1a7399587c493e3640.